### ISSUE TITLE
sandbox group can not be used normally

### ISSUE THREAD

**Issue by sv158** [sandboxie-plus/Sandboxie#1362]
**Describe the bug**

main problem: sandbox group name become `[none]` after creating new sandbox with existing sandbox's config.

Here are more details about:
- if you move new sandbox to one of those `[none]` group, first time it will clear the exist group. After that if you restart Sanboxie Plus, there is only one `[none]` group.
- I tested some cases and found the reason is that Sandboxie config will be reloaded after create new sandbox(no matter it init with existing sanbox's config), and the group changes haven't been save yet so it lost after reloaded.
- if I create a group and exit Sandboxie first, then create a sandbox and move it to the group, it will not disorder the group list.

**To Reproduce**
1. create a group named 'g1'. then we have N+1 groups in group list(count N≥0)
2. move an existing sandbox 'box1' group to 'g1'
3. create a new sandbox named 'box2'
4. see the group list changes

**Screenshots**
N/A

**Expected behavior**
- group list has N+1 groups, with 'g1' as the last one.
- after restart Sandboxie, there are still N+1 groups.

**System details and installed software**
 - OS version: Windows 10 Pro 21H1 64-bit, NT 10.0.19043.1288;
 - Sandboxie version: Sandboxie Plus 0.9.8d 64-bit;
 - Sandboxie version before the update: N/A, new system with first time installation of Sandboxie;
 - version this bug got introduced: I don't know. I used Sandboxie Plus before, but this is my first time use sandbox group;
 - security software running in the background: Kaspersky Free Anti-virus 21.3.10.191

**If you have a compatibility issue**
no related compatibility issue.

**Additional context**

**Sandboxie configuration**

<details>

<summary>My initial sandboxie.ini settings</summary>

```
[GlobalSettings]
FileRootPath=E:\Sandbox\%USER%\%SANDBOX%
SeparateUserFolders=y
KeyRootPath=\REGISTRY\USER\Sandbox_%USER%_%SANDBOX%
IpcRootPath=\Sandbox\%USER%\%SANDBOX%\Session_%SESSION%
NetworkEnableWFP=y
EditAdminOnly=n
ForceDisableAdminOnly=n
ForgetPassword=n
Template=WindowsRasMan
Template=WindowsLive
Template=VirtualDesktopManager
Template=OfficeLicensing
Template=Kaspersky

.....
```
complete shareable config: https://paste.opendev.org/show/bboBb1wZpUrVV8V9p1Dt/
</details>

**Comment by Coverlin** [comment]
> **Describe the bug**
> 
> main problem: sandox group name become `[none]` after creating new sandbox with existing sandbox's config.
> 
> Here are more details about:
> 
> * if you move new sandbox to one of those `[none]` group, first time it will clear the exist group. After that if you restart Sanboxie Plus, there is only one `[none]` group.
> * I tested some cases and found the reason is that Sandboxie config will be reloaded after create new sandbox(no matter it init with existing sanbox's config), and the group changes haven't been save yet so it lost after reloaded.
> * if I create a group and exit Sandboxie first, then create a sandbox and move it to the group, it will not disorder the group list.
> 

#1358

**Comment by isaak654** [comment]
This should be fixed with [1.0.5](https://github.com/sandboxie-plus/Sandboxie/releases/tag/1.0.5) build.
